signers/v2/v2.go

```diff
@@ -126,7 +126,10 @@ func (v *V2Signer) CreateSignable(req *http.Request, authHeaders map[string]stri
 
 	b.WriteString(req.Header.Get("X-Authorization-Timestamp"))
 
-	if bodyhash != "" && req.ContentLength > 0 {
+	// We cannot rely on the contentLength to be accurate. If either the body
+	// or the contentLength is not empty, we add a hash of the body and the
+	// content-type.
+	if bodyhash != "" || req.ContentLength > 0 {
```

Comment:
After consideration, a change would be in order, yes - but not here. This bodyhash is calculated in v2.go:177. We should sanitize the check there and remove the ContentLength check from here.